script.rb

```diff
@@ -0,0 +1,148 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield self[i]
+
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  # my_select
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_array = []
+    i = 0
+    until i == size
+      selected_array << self[i] if yield(self[i])
+      i += 1
+    end
+    selected_array
+  end
```

Comment:
- [ ] `my_select` is not working as the original `select` method, also is not passing all test cases.

### `select` is an Enumerable that has these characteristics:

- Returns a **new array** with the results that match the condition in the block
```
[1, 2, 3].select { |num| num > 2 }
# [3]
```

[Enumerable#select](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-select)